lms/static/js/learner_dashboard/models/course_card_model.js

```diff
@@ -220,6 +226,10 @@
                 updateCourseRun: function(courseRunKey) {
                     var selectedCourseRun = _.findWhere(this.get('course_runs'), {key: courseRunKey});
                     if (selectedCourseRun) {
+                        // Set the course run to the enrolled state
+                        _.each(this.context.course_runs, function(run) {
+                            if (run.key === selectedCourseRun.key) run.is_enrolled = true; // eslint-disable-line no-param-reassign, max-len
```

Comment:
Why are you disabling an eslint rule here instead of updating your code to conform?

Reply:
The two options were the ones below and I thought the second one looked a lot cleaner. 

The reason for that linter is to make sure that the developer is aware that they are altering the state of the actual object, which in this case is the behavior that we want. The solution that avoids this issue uses 'map' which creates a copy of the object and then reassigns it, adding to the data storage the function uses. Another solution uses a for loop which seems clunky. I can change it if you feel strongly but otherwise will leave as is.

![screen shot 2017-12-06 at 2 41 32 pm](https://user-images.githubusercontent.com/3861052/33681786-96342d7e-da93-11e7-95f1-81d7fa2b8fcb.png)

Reply:
That's fine but it would be great to add a comment whenever an eslint rule is disabled explaining the reasoning (to make it clearer for people working on the code in the future).

Reply:
Sounds good, I'll add a comment now